### PR TITLE
Add support for binary file request body

### LINKED ISSUE #455: [Feature] Support audio binary file type
### What Would You Like to See with the Gateway?

Add support for Audio to text transcription through a proxy for BYOLLM

Support this endpoint https://developers.deepgram.com/reference/listen-file

### Context for your Request

_No response_

### Your Twitter/LinkedIn

_No response_

### PR DESCRIPTION
**Pull request by narengogi**
**Title:** 
- Add support for binary file type through proxy

**Description:** (optional)
- There was 1 issue with provider check for proxy, removed that
- There was 1 null check missing inside the responseHandler method inside handlerutils, fixed that

**Related Issues:** (optional)
- https://github.com/Portkey-AI/gateway/issues/455

### CODE REVIEW

File: src/handlers/proxyHandler.ts [hunk 1]
```diff
@@ -206,13 +209,19 @@ export async function proxyHandler(c: Context): Promise<Response> {
       ) as Config | ShortConfig;
     }
 
+    let body;
+    if (requestContentType.startsWith(CONTENT_TYPES.GENERIC_AUDIO_PATTERN)) {
+      body = requestBinary;
+    } else if (requestContentType === CONTENT_TYPES.APPLICATION_JSON) {
+      body = store.requestFormData;
```
Comment (VisargD) [inline src/handlers/proxyHandler.ts:221]: Hey! This condition is incorrect. Can you please fix it?
The condition checks for application/json content type but the value being set for `body` is `store.requestFormData`.

Comment (narengogi) [reply src/handlers/proxyHandler.ts:221]: Updated, thanks for catching.
Moving the PR to draft to test again

**Comment by narengogi** [comment]
@VisargD Pending before release:
Test Brotli compression on Cloudflare and nodejs enterprise

File: src/middlewares/requestValidator/index.ts [hunk 2]
```diff
@@ -5,12 +5,16 @@ import { configSchema } from './schema/config';
...
+  const contentType = requestHeaders['content-type'];
   if (
-    requestHeaders['content-type'] &&
-    ![
+    !!contentType &&
+    (![
       CONTENT_TYPES.APPLICATION_JSON,
       CONTENT_TYPES.MULTIPART_FORM_DATA,
-    ].includes(requestHeaders['content-type'].split(';')[0])
+    ].includes(requestHeaders['content-type'].split(';')[0]) ||
```
Comment (VisargD) [inline src/middlewares/requestValidator/index.ts]: This nested OR condition looks incorrect. Its causing issues even when I send a valid content-type header

**Pull request merged**